patroni/postgresql.py

```diff
@@ -808,6 +808,10 @@ def wait_for_port_open(self, pid, initiated, timeout):
             if not self.is_pid_running(pid):
                 logger.error('postmaster is not running')
                 self.set_state('start failed')
+                # somehow we observed multiple times that there were no postmaster running, but
+                # pg_controldata was still reporting: `Database cluster state: in archive recovery`
+                # Therefore we will try to give some more time to all processes to finish.
+                time.sleep(1)
```

Comment:
Unfortunately it doesn't help.
The problem arises only when there are archive_command and recovery_command used in addition to streaming replication.
Or we can rephrase even more strict, recovery_command is a problem and breaks everything...